runtime/tests/integration_tests.rs

```diff
@@ -62,15 +65,55 @@ fn run_rt_test(test_bin_name: Option<&'static str>) -> DefaultHwModel {
     model
 }
 
-fn send_mailbox_command(model: &mut DefaultHwModel, cmd_id: u32, arg_buf: &[u32]) -> Vec<u32> {
+/// Stream `msg` to Caliptra's SHA hardware
+fn stream_to_sha(model: &mut DefaultHwModel, msg: &[u32]) {
+    model.soc_sha512_acc().control().write(|w| w.zeroize(true));
+
+    if model.soc_sha512_acc().lock().read().lock() {
+        panic!("sha_acc lock could not be acquired");
+    }
+
+    model
+        .soc_sha512_acc()
+        .dlen()
+        .write(|_| (msg.len() * 4).try_into().unwrap());
+
+    model
+        .soc_sha512_acc()
+        .mode()
+        .write(|w| w.mode(|w| w.sha_stream_384()));
+
+    for word in msg {
+        model.soc_sha512_acc().datain().write(|_| word.swap_bytes());
+    }
+
+    model.soc_sha512_acc().execute().write(|w| w.execute(true));
+
+    model.step_until(|m| m.soc_sha512_acc().status().read().valid());
+
+    model.soc_sha512_acc().lock().write(|w| w.lock(true)); // clear lock
+}
+
+/// Sends a SoC -> uC command to the mailbox
+///
+/// If the command succeeds, returns the response buffer. On error, returns the
+/// SoC ifc firmware error code.
+fn send_mailbox_command(
+    model: &mut DefaultHwModel,
+    cmd: CommandId,
+    cmd_buf: &[u8],
+) -> Result<Vec<u32>, u32> {
     model.soc_mbox().lock().read().lock();
-    model.soc_mbox().cmd().write(|_| cmd_id);
+    model.soc_mbox().cmd().write(|_| cmd as u32);
     model
         .soc_mbox()
         .dlen()
-        .write(|_| u32::try_from(arg_buf.len()).unwrap() * 4);
+        .write(|_| u32::try_from(cmd_buf.len()).unwrap());
 
-    for word in arg_buf {
+    let (prefix, words, suffix) = unsafe { cmd_buf.align_to::<u32>() };
```

Comment:
I don't think you need to hyper-optimize this with unsafe code. Use zerocopy, or use HwModel::mailbox_execute() directly.